File: .new/templates/java/Entity.template.java
/**
 * Entity Template for Java
 * 
 * Variables:
 *   ${package} - Package name
 *   ${entity_name} - Entity class name (PascalCase)
 *   ${id_type} - ID type (default: UUID)
 *   ${entity_base} - Base class for entities
 *   ${properties} - Property definitions
 *   ${constructor_params} - Constructor parameters
 *   ${methods} - Business methods
 */

package ${package}.domain.entities;

import java.util.*;
import ${package}.domain.shared.*;
import ${package}.domain.events.*;

/**
 * ${entity_description}
 */
public class ${entity_name} extends ${entity_base}<${id_type}> {
    
    // ============================================
    // Properties
    // ============================================
    
    ${properties}
    
    // ============================================
    // Constructors
    // ============================================
    
    /**
     * Creates a new {@link ${entity_name}} instance.
     */
    public ${entity_name}(${constructor_params}) {
        super(${id_assignment});
        
        // Validate invariants
        ${constructor_validations}
        
        // Set properties
        ${constructor_assignments}
    }
    
    /**
     * Protected constructor for JPA.
     */
    protected ${entity_name}() {
        super();
    }
    
    // ============================================
    // Getters
    // ============================================
    
    ${getters}
    
    // ============================================
    // Business Methods
    // ============================================
    
    ${methods}
    
    // ============================================
    // Domain Events
    // ============================================
    
    ${domain_events}
}

/*
 * Property Template:
 * 
 * private ${property_type} ${property_name};
 */

/*
 * Getter Template:
 * 
 * /**
 *  * Gets the ${property_description}.
 *  */
 * public ${property_type} get${PropertyName}() {
 *     return ${property_name};
 * }
 */

/*
 * Method Template:
 * 
 * /**
 *  * ${method_description}
 *  */
 * public ${return_type} ${method_name}(${method_params}) {
 *     // Validate preconditions
 *     ${method_validations}
 *     
 *     // Perform operation
 *     ${method_body}
 *     
 *     // Raise domain event if needed
 *     registerDomainEvent(new ${event_name}(this));
 * }
 */

/*
 * Validation Template:
 * 
 * Objects.requireNonNull(${param_name}, "${param_name} cannot be null");
 * 
 * if (${condition}) {
 *     throw new IllegalArgumentException("${message}");
 * }
 */

File: .new/templates/dotnet/entity.template.cs
/*
 * Entity Template for .NET
 * 
 * Variables:
 *   ${namespace} - Project namespace
 *   ${entity_name} - Entity class name (PascalCase)
 *   ${id_type} - ID type (default: Guid)
 *   ${entity_base} - Base class for entities
 *   ${properties} - Property definitions
 *   ${constructor_params} - Constructor parameters
 *   ${methods} - Business methods
 */

using System;
using System.Collections.Generic;
using ${namespace}.Domain.Shared;

namespace ${namespace}.Domain.Entities
{
    /// <summary>
    /// ${entity_description}
    /// </summary>
    public class ${entity_name} : ${entity_base}<${id_type}>
    {
        #region Properties
        
        ${properties}
        
        #endregion
        
        #region Constructors
        
        /// <summary>
        /// Creates a new <see cref="${entity_name}"/> instance.
        /// </summary>
        public ${entity_name}(${constructor_params})
        {
            // Validate invariants
            ${constructor_validations}
            
            // Set properties
            ${constructor_assignments}
        }
        
        /// <summary>
        /// Private constructor for ORM.
        /// </summary>
        private ${entity_name}() { }
        
        #endregion
        
        #region Business Methods
        
        ${methods}
        
        #endregion
        
        #region Domain Events
        
        ${domain_events}
        
        #endregion
    }
}

/*
 * Property Template:
 * 
 * /// <summary>
 * /// ${property_description}
 * /// </summary>
 * public ${property_type} ${property_name} { get; private set; }
 */

/*
 * Method Template:
 * 
 * /// <summary>
 * /// ${method_description}
 * /// </summary>
 * public ${return_type} ${method_name}(${method_params})
 * {
 *     // Validate preconditions
 *     ${method_validations}
 *     
 *     // Perform operation
 *     ${method_body}
 *     
 *     // Raise domain event if needed
 *     AddDomainEvent(new ${event_name}(this));
 * }
 */

/*
 * Validation Template:
 * 
 * if (${condition})
 * {
 *     throw new ArgumentException("${message}", nameof(${param_name}));
 * }
 */
